test/integration_test.py

```diff
@@ -714,3 +724,51 @@ def test_mini_dss_failures(self):
                 mini_dss.get_file(uuid, version, 'aws')
             with self.assertRaises(self.SpecialError):
                 mini_dss.get_native_file_url(uuid, version, 'aws')
+
+
+class AzulChaliceLocalIntegrationTest(AzulTestCase):
+    host = '127.0.0.1'
+    port = 8000
+    url = furl(scheme='http', host=host, port=port)
+    local_chalice_server = None
+    thread = None
+
+    @classmethod
+    def setUpClass(cls, config=config) -> None:
+        super().setUpClass()
+        service_app = Path(config.project_root) / 'lambdas' / 'service'
+        localfactory = factory.CLIFactory(str(service_app))
+        config = localfactory.create_config_obj()
+        cls.local_chalice_server = localfactory.create_local_server(app_obj=app,
+                                                                    config=config,
+                                                                    host=cls.host,
+                                                                    port=cls.port)
+        cls.thread = threading.Thread(target=cls.local_chalice_server.serve_forever)
+        cls.thread.start()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.local_chalice_server.shutdown()
+        cls.thread.join()
+        super().tearDownClass()
+
+    def test_local_chalice_health_endpoint(self):
+        url = self.url
+        url.set(path='health')
+        response = requests.get(url.url)
+        self.assertEqual(response.status_code, 200)
+
+    catalog = first(config.integration_test_catalogs.keys())
+
+    def test_local_chalice_index_endpoints(self):
+        url = self.url
+        url.set(path='index/files', query=dict(catalog=self.catalog))
+        response = requests.get(url.url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_local_filtered_index_endpoints(self):
+        filters = {"genusSpecies": {"is": ["Homo sapiens"]}}
+        url = self.url
+        url.set(path='index/files', query=dict(filters={json.dumps(filters)}, catalog=self.catalog))
```

Comment:
Modifies self.url and introduces sensitivity to the order in which tests are executed.